crates/apub/src/activities/following/follow.rs

```diff
@@ -102,21 +102,25 @@ impl ActivityHandler for Follow {
           pending: false,
         };
         PersonFollower::follow(&mut context.pool(), &form).await?;
+        AcceptFollow::send(self, context).await?;
       }
       UserOrCommunity::Community(c) => {
-        // Dont allow following local-only community via federation.
-        if c.visibility != CommunityVisibility::Public {
-          return Err(LemmyErrorType::NotFound.into());
-        }
+        let state = Some(match c.visibility {
+          CommunityVisibility::Public => CommunityFollowerState::Accepted,
+          CommunityVisibility::Private => CommunityFollowerState::ApprovalRequired,
+          // Dont allow following local-only community via federation.
+          CommunityVisibility::LocalOnly => return Err(LemmyErrorType::NotFound.into()),
+        });
         let form = CommunityFollowerForm {
-          community_id: c.id,
-          person_id: actor.id,
-          pending: false,
+          state,
+          ..CommunityFollowerForm::new(c.id, actor.id)
```

Comment:
I commented above about doing this, it probably could be extracted, or made into an impl function of either CommunityVisibility, or CommunityFollowerState, or just into a utils crate.